src/hardware/disney.cpp

```diff
@@ -98,6 +98,19 @@ static void DISNEY_enable(uint32_t freq)
 	disney.state = STATE::RUNNING;
 }
 
+// Calculate the frequency from DAC samples and speed parameters
+// The maximum possible frequency return is 127000 Hz, which
+// occurs when all 128 samples are used and the speedcheck_sum
+// has accumulated only one single tick.
+static uint32_t calc_frequency(const dac_channel &dac)
+{
+	if (dac.used <= 1)
+		return 0;
+	const uint32_t k_samples = 1000 * (dac.used - 1);
+	const auto frequency = k_samples / dac.speedcheck_sum;
```

Comment:
Maybe we want an assertion in here to protect from `dac.speedcheck_sum` == 0.0, which would make `frequency` floating-point inf?

Or maybe this should be a runtime behaviour instead?

Previously code had comment indicating that check on `dac.used` implies protection from division by zero, but now it's disconnected.

Reply:
`speedcheck_sum` has an existing runtime check:

``` c++
			cch->speedcheck_sum += current - cch->speedcheck_last;

			if((current - cch-> speedcheck_last) < 0.01 ||
				(current - cch-> speedcheck_last) > 2)
				cch->speedcheck_failed = true;
			
			// if both are failed we are back at start
			if(disney.da[0].speedcheck_failed && disney.da[1].speedcheck_failed) {
				disney.state=STATE::IDLE;
				break;
			}
```

`speedcheck_sum` can only be used when `STATE::FINISHED`, and the only way to get the state to `disney.state = STATE::FINISHED;` is to get passed that chunk of check code.